drivers/sensor/iis2mdc/iis2mdc_spi.c

```diff
@@ -83,45 +81,38 @@ static int iis2mdc_spi_write(struct iis2mdc_data *data, uint8_t reg_addr,
 		.count = 2
 	};
 
-
 	if (len > 64) {
 		return -EIO;
 	}
 
-	if (spi_write(data->bus, spi_cfg, &tx)) {
+	if (spi_write(cfg->bus, spi_cfg, &tx)) {
 		return -EIO;
 	}
 
 	return 0;
 }
 
+stmdev_ctx_t iis2mdc_spi_ctx = {
+	.read_reg = (stmdev_read_ptr) iis2mdc_spi_read,
+	.write_reg = (stmdev_write_ptr) iis2mdc_spi_write,
+};
+
 int iis2mdc_spi_init(const struct device *dev)
 {
 	struct iis2mdc_data *data = dev->data;
+	const struct iis2mdc_dev_config *const cfg = dev->config;
+	const struct spi_cs_control *cs_ctrl = cfg->bus_cfg.spi_cfg.cs;
 
-	data->ctx_spi.read_reg = (stmdev_read_ptr) iis2mdc_spi_read;
-	data->ctx_spi.write_reg = (stmdev_write_ptr) iis2mdc_spi_write;
-
-	data->ctx = &data->ctx_spi;
-	data->ctx->handle = data;
-
-#if DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
-	const struct iis2mdc_config *cfg = dev->config;
-
-	/* handle SPI CS thru GPIO if it is the case */
-	data->cs_ctrl.gpio_dev = device_get_binding(cfg->gpio_cs_port);
-	if (!data->cs_ctrl.gpio_dev) {
-		LOG_ERR("Unable to get GPIO SPI CS device");
+	if (!device_is_ready(cs_ctrl->gpio_dev)) {
+		LOG_ERR("Cannot get pointer to CS gpio device");
 		return -ENODEV;
 	}
 
-	data->cs_ctrl.gpio_pin = cfg->cs_gpio;
-	data->cs_ctrl.gpio_dt_flags = cfg->cs_gpio_flags;
-	data->cs_ctrl.delay = 0;
+	data->ctx_spi.read_reg = (stmdev_read_ptr) iis2mdc_spi_read;
+	data->ctx_spi.write_reg = (stmdev_write_ptr) iis2mdc_spi_write;
 
-	LOG_DBG("SPI GPIO CS configured on %s:%u",
-		cfg->gpio_cs_port, cfg->cs_gpio);
-#endif
+	data->ctx = &data->ctx_spi;
```

Comment:
Should this be:

    data->ctx = &iis2mdc_spi_ctx;

?

Reply:
@pabigot 
No, I just removed the iis2mdc_spi_ctx because the ctx structure is directly inside data. But is like this since long back, and all structures like iis2mdc_spi_ctx and iis2mdc_i2c_ctx are not needed anymore. I'm not sure how it happened that in this PR at a certain point I brought in again this structure, but at least I learned the term "stray addition". Thanks for catching it.
BTW, in ii2smdc I still need to use the stmemsc coomon routines introduced in #33506. Moreover, I'm considering moving the ctx inside stmemsc_cfg_xxx structures, so that I should be able to completely remove the ST_DRVXXX_i2c.c and ST_DRVXXX_spi.c files completely.